File: backend/src/models/OrderCategory.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IOrderCategory extends Document {
  name: string;
  description?: string;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const OrderCategorySchema: Schema<IOrderCategory> = new Schema(
  {
    name: {
      type: String,
      required: [true, 'Order category name is required'],
      trim: true,
      unique: true,
    },
    description: {
      type: String,
      trim: true,
      default: '',
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

OrderCategorySchema.index(
  { name: 1 },
  {
    unique: true,
    collation: {
      locale: 'en',
      strength: 2,
    },
  }
);

export default mongoose.model<IOrderCategory>('OrderCategory', OrderCategorySchema);

File: user-app/lib/models/order_category.dart
class OrderCategory {
  final String id;
  final String name;
  final String description;
  final bool isActive;

  const OrderCategory({
    required this.id,
    required this.name,
    required this.description,
    required this.isActive,
  });

  factory OrderCategory.fromJson(Map<String, dynamic> json) {
    return OrderCategory(
      id: json['_id']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      isActive: json['isActive'] == true,
    );
  }
}
